src/sparse_tabular.jl

```diff
@@ -0,0 +1,156 @@
+struct SparseTabularMDP <: MDP{Int64, Int64}
+    T::Vector{SparseMatrixCSC{Float64, Int64}} # T[a][s, sp]
+    R::Array{Float64, 2} # R[s,sp]
+    discount::Float64
+end
+
+function SparseTabularMDP(mdp::MDP)
+    T = transition_matrix_a_s_sp(mdp)
+    R = reward_s_a(mdp)
+    return SparseTabularMDP(T, R, discount(mdp))
+end
+
+function SparseTabularMDP(mdp::SparseTabularMDP;
```

Comment:
It translates the problem to one where the states and actions are integers. This could be useful because you may want to write some really fast solver that is optimized to assume all states are integers. If the user gives this solver a problem that does not have integer states, it can use this representation to handle it. I think it is potentially useful.

Reply:
I think @rejuvyesh was referring to the other constructor that takes ::SparseTabularMDP as input. This is if someone wants to change the reward matrix but keep the transition because it takes a while to compute. This is the copy constructor mentioned in that issue: JuliaPOMDP/DiscreteValueIteration.jl#24 . 
I do agree that it is not a big deal to just call the default constructor directly instead.

Reply:
Do you think that we should provide a function to access the transition and observation matrices or is it sufficient to document it in a docstring in the `SparseTabular` type definition? 

otherwise something like `transition_matrix(mdp::SparseTabularMDP) = mdp.T`

Reply:
Yes! I think functions are much better than accessing fields directly. My recommendations would be
```julia
transition_matrix(m::SparseTabularProblem, a)
```
and
```julia
reward_vector(m::SparseTabularProblem, a)
```
Those are what would be needed for VI https://github.com/JuliaPOMDP/DiscreteValueIteration.jl/blob/master/src/sparse.jl#L44

Reply:
Note however, these won't allow the user to switch between indexing and view mode.

Reply:
It's better to just have 
```julia
transition_matrix(m::SparseTabularProblem)
```
and
```julia
reward_vector(m::SparseTabularProblem)
```

Reply:
These are different methods so technically we could have both